source/dubregistry/registry.d

```diff
@@ -248,6 +250,44 @@ class DubRegistry {
 		return ret;
 	}
 
+	PackageInfo[string] getPackageInfoWithDependencies(string packname, bool include_errors, bool minimize)
+	{
+		PackageInfo[string] ret;
+		getPackageInfoWithDependencies(packname, include_errors, minimize, ret);
+		logDebug("%s", ret.keys);
+		return ret;
+	}
+
+	void getPackageInfoWithDependencies(string packname, bool include_errors, bool minimize, ref PackageInfo[string] ret)
+	{
+		static string rootOf(string depname)
+		{
+			import std.algorithm: findSplitBefore;
+			return depname.findSplitBefore(":")[0];
+		}
+
+		logDebug("getPackageInfoWithDependencies: %s", packname);
+		auto p = ret[packname] = getPackageInfo(packname, include_errors, minimize);
+		foreach (v; p.versions)
+		{
+			foreach (dep, _; v.info["dependencies"].opt!(Json[string]))
+			{
+				auto depname = rootOf(dep);
+				if (depname !in ret)
+					getPackageInfoWithDependencies(depname, include_errors, minimize, ret);
+			}
+			foreach (cfg; v.info["configurations"].opt!(Json[]))
+			{
+				foreach (dep, _; v.info["dependencies"].opt!(Json[string]))
```

Comment:
This looks wrong. Shouldn't it be `cfg.info`?

Reply:
Yes, fixed